src/components/data-management/AddMetadataButton.jsx

```diff
@@ -17,31 +21,52 @@ const AddMetadataButton = ({ samplesTableRef }) => {
   const isSubsetted = activeExperiment?.isSubsetted;
   const samples = useSelector((state) => state.samples);
   const selectedTech = samples[activeExperiment?.sampleIds[0]]?.type;
-  const [uploadModalVisible, setUploadModalVisible] = useState(false);
 
-  const uploadFiles = (file) => {
+  const [uploadModalVisible, setMetadataUploadModalVisible] = useState(false);
+  const [cellLevelUploadVisible, setCellLevelUploadVisible] = useState(false);
+
+  const onUploadMetadataTrack = (file) => {
     dispatch(uploadMetadataFile(activeExperimentId, file));
-    setUploadModalVisible(false);
+    setMetadataUploadModalVisible(false);
+  };
+
+  const onUploadCellLevelMetadata = async (file) => {
+    const body = {
+      name: file.name,
+    };
+
+    const signedUploadURL = await dispatch(createCellLevelMetadata(activeExperimentId, body));
+    await putPartInS3(file, signedUploadURL, (progressEvent) => console.log('uploaded ', progressEvent.loaded, ' of total ', progressEvent.total));
+    setCellLevelUploadVisible(false);
```

Comment:
What do you think about perhaps making the modal be only a loader while the upload is ongoing?

Reply:
Also, if the upload fails, perhaps we could push an `endUserMessage`?